tests fail with thousands of "Assertion failure in -[NSMenu _lockForMainMenuItemArray], NSMenu.m:1106"

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Run `go test ./...`. Result is thousands (or more, I didn't wait for it to stop spewing) of lines like:

```
2024-01-27 07:58:05.095 glfw.test[33753:5608525] *** Assertion failure in -[NSMenu _lockForMainMenuItemArray], NSMenu.m:1106
```



### How to reproduce

See above.

### Screenshots

_No response_

### Example code

Again, see above.

### Fyne version

caf50f6407d7f8a5abc91064338ce9bc3b1820bb

### Go compiler version

1.21.6

### Operating system and version

macOS Sonoma

### Additional Information

I would be happy to try my hand at a fix, but I don't see any clear hints from grepping the source code about where to even start looking.

For my future reference, here's the tail of the failure:

```
[lots of these lines]
2024-01-27 15:56:26.282 glfw.test[63243:5971505] *** Assertion failure in -[NSMenu _lockForMainMenuItemArray], NSMenu.m:1106
panic: test timed out after 10m0s
running tests:
	TestDarwinMenu (9m56s)

goroutine 121 [running]:
testing.(*M).startAlarm.func1()
	/Users/josh/go/1.21/src/testing/testing.go:2259 +0x320
created by time.goFunc
	/Users/josh/go/1.21/src/time/sleep.go:176 +0x38

goroutine 1 [select, locked to thread]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).runGL(0x140000fe180)
	/Users/josh/x/fyne/internal/driver/glfw/loop.go:120 +0x14c
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run(0x140000fe180)
	/Users/josh/x/fyne/internal/driver/glfw/driver.go:165 +0x78
fyne.io/fyne/v2/internal/driver/glfw.TestMain(0x140000a1ae0)
	/Users/josh/x/fyne/internal/driver/glfw/window_test.go:53 +0x9c
main.main()
	_testmain.go:219 +0x1ac

goroutine 6 [chan receive, 9 minutes]:
fyne.io/fyne/v2/test.NewApp.func1()
	/Users/josh/x/fyne/test/testapp.go:167 +0x44
created by fyne.io/fyne/v2/test.NewApp in goroutine 1
	/Users/josh/x/fyne/test/testapp.go:165 +0x334

goroutine 7 [select, 9 minutes, locked to thread]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread.func1()
	/Users/josh/x/fyne/internal/driver/glfw/loop.go:240 +0xa8
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).startDrawThread in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/loop.go:236 +0xc4

goroutine 8 [chan receive, 9 minutes]:
testing.(*T).Run(0x140001029c0, {0x104534205?, 0x1f30457a96a26?}, 0x1046de678)
	/Users/josh/go/1.21/src/testing/testing.go:1649 +0x350
testing.runTests.func1(0x140001029c0?)
	/Users/josh/go/1.21/src/testing/testing.go:2054 +0x48
testing.tRunner(0x140001029c0, 0x14000066d28)
	/Users/josh/go/1.21/src/testing/testing.go:1595 +0xe8
testing.runTests(0x140000a1ae0?, {0x104958fe0, 0x53, 0x53}, {0x14000066e38?, 0x104176e28?, 0x105082980?})
	/Users/josh/go/1.21/src/testing/testing.go:2052 +0x3b4
testing.(*M).Run(0x140000a1ae0)
	/Users/josh/go/1.21/src/testing/testing.go:1925 +0x538
fyne.io/fyne/v2/internal/driver/glfw.TestMain.func1()
	/Users/josh/x/fyne/internal/driver/glfw/window_test.go:46 +0x40
created by fyne.io/fyne/v2/internal/driver/glfw.TestMain in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window_test.go:40 +0x6c

goroutine 9 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 10 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 18 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).catchTerm(0x0?)
	/Users/josh/x/fyne/internal/driver/glfw/driver_desktop.go:182 +0x7c
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).Run in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/driver.go:164 +0x70

goroutine 19 [syscall, 9 minutes]:
os/signal.signal_recv()
	/Users/josh/go/1.21/src/runtime/sigqueue.go:149 +0x2c
os/signal.loop()
	/Users/josh/go/1.21/src/os/signal/signal_unix.go:23 +0x1c
created by os/signal.Notify.func1.1 in goroutine 18
	/Users/josh/go/1.21/src/os/signal/signal.go:151 +0x28

goroutine 14 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x1400011c030?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 13 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 143 [runnable]:
fyne.io/fyne/v2/internal/driver/glfw._Cfunc_test_NSMenu_removeItemAtIndex(0x600003167700, 0x1)
	_cgo_gotypes.go:316 +0x30
fyne.io/fyne/v2/internal/driver/glfw.resetMainMenu.testNSMenuRemoveItemAtIndex.func4(0x600003167700?, 0x1?)
	/Users/josh/x/fyne/internal/driver/glfw/menu_darwin.go:324 +0x50
fyne.io/fyne/v2/internal/driver/glfw.testNSMenuRemoveItemAtIndex(...)
	/Users/josh/x/fyne/internal/driver/glfw/menu_darwin.go:324
fyne.io/fyne/v2/internal/driver/glfw.resetMainMenu()
	/Users/josh/x/fyne/internal/driver/glfw/menu_darwin_test.go:300 +0xec
fyne.io/fyne/v2/internal/driver/glfw.TestDarwinMenu(0x14000b4c820)
	/Users/josh/x/fyne/internal/driver/glfw/menu_darwin_test.go:19 +0xa8
testing.tRunner(0x14000b4c820, 0x1046de678)
	/Users/josh/go/1.21/src/testing/testing.go:1595 +0xe8
created by testing.(*T).Run in goroutine 8
	/Users/josh/go/1.21/src/testing/testing.go:1648 +0x33c

goroutine 16 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 38 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 39 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 51 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 52 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 54 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 55 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 41 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 42 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 57 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 58 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 60 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 61 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 67 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 68 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 46 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 47 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 82 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x14000303040?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 98 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 99 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 73 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 74 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 84 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 85 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 49 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 114 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 116 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 117 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 119 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 120 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 87 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 88 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 90 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 91 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 93 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 94 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 147 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 100 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 101 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 130 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 131 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 76 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 77 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 79 [select, 9 minutes]:
fyne.io/fyne/v2/internal/async.(*UnboundedChan[...]).processing(0x1046e98c0)
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:74 +0xb4
created by fyne.io/fyne/v2/internal/async.NewUnboundedChan[...] in goroutine 1
	/Users/josh/x/fyne/internal/async/chan_go1.21.go:51 +0x10c

goroutine 80 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150

goroutine 148 [chan receive, 9 minutes]:
fyne.io/fyne/v2/internal/driver/common.(*Window).RunEventQueue(0x0?)
	/Users/josh/x/fyne/internal/driver/common/window.go:34 +0x48
created by fyne.io/fyne/v2/internal/driver/glfw.(*gLDriver).createWindow.func1 in goroutine 1
	/Users/josh/x/fyne/internal/driver/glfw/window.go:957 +0x150
FAIL	fyne.io/fyne/v2/internal/driver/glfw	602.334s
```

--- a/internal/driver/glfw/menu_darwin_test.go
+++ b/internal/driver/glfw/menu_darwin_test.go
@@ -16,7 +16,9 @@ func TestDarwinMenu(t *testing.T) {
 	setExceptionCallback(func(msg string) { t.Error("Obj-C exception:", msg) })
 	defer setExceptionCallback(nil)
 
-	resetMainMenu()
+	runOnMain(func() {
+		resetMainMenu()
+	})
 
 	w := createWindow("Test").(*window)
 
@@ -58,7 +60,9 @@ func TestDarwinMenu(t *testing.T) {
 	menuSettings := fyne.NewMenu("Settings", itemSettings, fyne.NewMenuItemSeparator(), itemMoreSetings)
 
 	mainMenu := fyne.NewMainMenu(menuEdit, menuHelp, menuMore, menuSettings)
-	setupNativeMenu(w, mainMenu)
+	runOnMain(func() {
+		setupNativeMenu(w, mainMenu)
+	})
 
 	mm := testDarwinMainMenu()
 	// The custom “Preferences” menu should be moved to the system app menu completely.
@@ -246,13 +250,17 @@ func TestDarwinMenu_specialKeyShortcuts(t *testing.T) {
 		},
 	} {
 		t.Run(name, func(t *testing.T) {
-			resetMainMenu()
+			runOnMain(func() {
+				resetMainMenu()
+			})
 			w := createWindow("Test").(*window)
 			item := fyne.NewMenuItem("Special", func() {})
 			item.Shortcut = &desktop.CustomShortcut{KeyName: tt.key, Modifier: fyne.KeyModifierShortcutDefault}
 			menu := fyne.NewMenu("Special", item)
 			mainMenu := fyne.NewMainMenu(menu)
-			setupNativeMenu(w, mainMenu)
+			runOnMain(func() {
+				setupNativeMenu(w, mainMenu)
+			})
 
 			mm := testDarwinMainMenu()
 			m := testNSMenuItemSubmenu(testNSMenuItemAtIndex(mm, 1))

